src/containers/pages/IRS/assignments/index.tsx

```diff
@@ -78,6 +78,7 @@ const IRSAssignmentPlansList = (props: PlanAssignmentsListProps) => {
   /** async function to load the data */
   async function loadData() {
     try {
+      setLoading(plans.length < 1); // only set loading when there are no plans
```

Comment:
Why use `plans.length < 1` as the test here?  iirc we are trying to move away from that, no?

Reply:
Ooh yeah, this is a stoaway commit when branching of. The changes are only for practitioners list view

Reply:
Cool, cool

Reply:
How can we make this so generic that when we implement this when fetching plans, or jurisdictions or whatever else (from OpenSRP for now), we don't have to re-write similar(ish) code?

Reply:
hmm,...

Reply:
> How can we make this so generic that when we implement this when fetching plans, or jurisdictions or whatever else (from OpenSRP for now), we don't have to re-write similar(ish) code?

@moshthepitt  So just to be clear, the quoted text above is about the react-async changes introduced in the practitionerListView and not in the IRS file of which this thread is part of, ama?

Reply:
Not sure what you mean, but what I mean is the usage of react-async to combat the problems we identified in #806 should be as DRY as possible across different components.

Reply:
ooh!,.. ok.